gnuradio-runtime/lib/buffer.cc

```diff
@@ -263,92 +216,57 @@ void buffer::prune_tags(uint64_t max_time)
     }
 }
 
-long buffer_ncurrently_allocated() { return s_buffer_count; }
-
-// ----------------------------------------------------------------------------
-
-buffer_reader::buffer_reader(buffer_sptr buffer, unsigned int read_index, block_sptr link)
-    : d_buffer(buffer),
-      d_read_index(read_index),
-      d_abs_read_offset(0),
-      d_link(link),
-      d_attr_delay(0)
+void buffer::on_lock(gr::thread::scoped_lock& lock)
 {
-    s_buffer_reader_count++;
+    // NOTE: the protecting mutex (scoped_lock) is held by the custom_lock object
```

Comment:
This mechanism could probably use some detailed comments on what the purpose is and how it functions

Reply:
Agreed, I will add more detailed comments.  The whole custom lock thing is unfortunately both complex and special purpose.